server/game/gameSystems/ReadyResourcesSystem.ts

```diff
@@ -0,0 +1,41 @@
+import { AbilityContext } from '../core/ability/AbilityContext';
+import type { Card } from '../core/card/Card';
+import { CardType, Location, WildcardCardType } from '../core/Constants';
+import Player from '../core/Player.js';
+import * as EnumHelpers from '../core/utils/EnumHelpers';
+import { IPlayerTargetSystemProperties, PlayerTargetSystem } from '../core/gameSystem/PlayerTargetSystem.js';
+
+
+export interface IReadyResourcesSystemProperties extends IPlayerTargetSystemProperties {
+    amount?: number
+}
+
+export class ReadyResourcesSystem<TContext extends AbilityContext = AbilityContext, TProperties extends IReadyResourcesSystemProperties = IReadyResourcesSystemProperties> extends PlayerTargetSystem<TContext, TProperties> {
+    protected override defaultProperties: IReadyResourcesSystemProperties = {
+        amount: 1
+    };
+
+    public override eventHandler(event): void {
+        event.player.readyResources(event.amount);
+    }
+
+    public override getEffectMessage(context: TContext): [string, any[]] {
+        const properties = this.generatePropertiesFromContext(context);
+        const message = 'ready {0} resources';
+        return [message, [properties.amount]];
+    }
+
+    public override canAffect(player: Player, context: TContext, additionalProperties = {}): boolean {
+        return player.countExhaustedResources() > 0;
```

Comment:
This check is slightly off, because an effect technically can target a player with no exhausted resources, it just won't do anything. However, if the effect is a cost, this must be true.

Take a look at how the `ReadySystem` handles `isCost` in this check as an example

Reply:
I based a lot of this class off of SearchDeckSystem(because it also targets a player), and the canAffect() in that class is only true if it could have an effect. Should that be changed as well?

Reply:
Yeah that's incorrect as well then, probably it's just that search deck isn't currently a cost for anything so it hasn't come up. We can change it in this PR or just add a TODO comment there.